vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -70,6 +71,13 @@ import { ComboBoxPlaceholder } from '@vaadin/combo-box/src/vaadin-combo-box-plac
         };
 
         comboBox.dataProvider = function (params, callback) {
+          const filterUnchanged = params.filter === lastFilter;
+          if (filterUnchanged && this._debouncer) {
+            if (pageInDebouncer !== params.page) {
+              pageCallbacks[params.page] = callback;
+            }
+            return;
+          }
```

Comment:
I'm wondering if it is necessary to store and check `pageInDebouncer`. The page that was requested first with the new filter can't be re-requested because it is already in the web component's `_pendingRequests`, I think. So, I wonder if we could just do something like that:

```js
if (filterChanged) {
  cache = {};
  lastFilter = params.filter;
  ...
}

if (this._filterDebouncer) {
  pageCallbacks[params.page] = callback;
  return;
}
```

Reply:
Addressed in new commit